### PR TITLE
Add option for timestamps from output of `Node.get_logs`

### PR DESCRIPTION
**Pull request by charlesbluca**
As discussed in #4909, it would be nice if the output of `Node.get_logs` had optional timestamps, which could be used to filter the logs in the performance report and/or be added as information for the user.

- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`

### CODE REVIEW

File: distributed/node.py
Comment (fjetter) [inline distributed/node.py:90]: It would be great if you can start off with a doc string for this. Start simple with the changes you contributed. We can add on to it later on.

File: distributed/dashboard/components/scheduler.py
Comment (fjetter) [inline distributed/dashboard/components/scheduler.py:2170]: We could, of course, also add a start/end kwarg to the get_logs and do the filtering in one pass.

**Review by fjetter — COMMENTED** [review]
CI failures are introduced since the performance report tests assert very specifically how the logs look like. You might want to relax the asserts over there.

**Comment by charlesbluca** [comment]
Are there any other user-facing places where we might want to add timestamps for logs? Currently this only impacts the perf report and Jupyter notebook representation.

**Comment by charlesbluca** [comment]
Will probably want to wait on #4941 before we merge this

**Comment by charlesbluca** [comment]
Could we rerun tests on this? Should be good now that #4941 is merged

File: distributed/node.py
Comment (fjetter) [inline distributed/node.py:119]: For performance, I would suggest to do this all in one go. how about

```python
if start is None:
    start = -1
L = []
for count, msg in enumerate(deque_handler.deque):
    if n and count >= n or msg.created < start:
        break
    if timestamps:
        L.append((msg.created, msg.levelname, deque_handler.format(msg)))
    else:
        L.append((msg.levelname, deque_handler.format(msg)))
return L
```

This way we can do everything in one pass instead of three. by not using a list comprehension we can also short-circuit the `start`since the logs should be always sorted. (haven't tested this)

**Review by fjetter — COMMENTED** [review]
LGTM. Please let me know if you want to try my suggestion or whether we shall merge

File: distributed/node.py
Comment (charlesbluca) [reply distributed/node.py:119]: This looks good - I will try this out

**Comment by charlesbluca** [comment]
Thanks for the suggestion @fjetter, think this should be good to merge now

**Review by fjetter — APPROVED** [review]
Thanks @charlesbluca and sorry for the long wait

**Pull request merged**
